### DISCUSSION TITLE
TLS certificate cache

### DISCUSSION THREAD

**Discussion by jsonn** [distribution/distribution#4288] [Q&A]
I'm using a container image build from Dockerfile and passing in a TLS certificate and key via REGISTRY_HTTP_TLS_CERTIFICATE and REGISTRY_HTTP_TLS_KEY respectively. Sometimes, access will be refused as a severely outdated version of the cert is used and I have no idea where it is coming from. This can be triggered with a plain curl/wget invocation, no docker/podman necessary. A consecutive invocation will suddenly work. Is there some cache I should know about?

**Comment by milosgajdos** [comment]
> Sometimes, access will be refused as a severely outdated version of the cert is used and I have no idea where it is coming from. This can be triggered with a plain curl/wget invocation, no docker/podman necessary. A consecutive invocation will suddenly work.

Mind dumping the cert using something like `openssl s_client -connect host:port`? 

Also, any logs would be useful though if the TLS connection fails to be established I struggle to see how any logs could be generated, still...
